tests/plugins/test_resolve_composes.py

```diff
@@ -1365,3 +1366,21 @@ def get_override_yum_repourls(self, workflow, arch=ODCS_COMPOSE_DEFAULT_ARCH):
                 .get(WORKSPACE_KEY_OVERRIDE_KWARGS, {})
                 .get(arch, {})
                 .get('yum_repourls'))
+
+    def test_skip_adjust_inheritance_for_composes_if_image_is_based_on_scratch(self, workflow):
+        plugin = ResolveComposesPlugin(workflow.builder.tasker, workflow)
+        flexmock(plugin.log).should_call('debug').once().with_args(
+            'This is a base image based on scratch. '
+            'Skipping forward to adjust inheritance for composes.'
+        )
+        flexmock(plugin.workflow.builder, base_from_scratch=True)
```

Comment:
we can just set it plugin.workflow.builder.base_from_scratch=True
we don't need flexmock for that